### ISSUE TITLE
Remove empty lines between roxygen docs and code?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/styler#936]
AFAIK, there are currently no guidelines about this in the _tidyverse style guide_, but I was wondering if {styler} should adopt a convention to remove empty lines between roxygen documentation and the code.

- **Actual**

In the wild, I often come across codebases that have formatting like the following. The _count_ of empty spurious lines may differ across files. Currently, {styler} preserves such empty lines.

``` r
styler::style_text(
  "#' @export
  


  foo <- function() NULL"
)
#> #' @export
#> 
#> 
#> 
#> foo <- function() NULL
```

- **Expected new behavior**

I wish {styler} could remove such inconsistencies and adopt a single rule. I'd propose that it removes all empty lines by default, but I am also open to other rules (e.g. always have a single empty line). The point is more about consistency in formatting.

```r
#> #' @export
#> foo <- function() NULL
``` 

- **Exception**

Just like in formatting `R6` lists, I think one exception to this rule could be the presence of a comment block before the function. In such cases, a single empty line can be preserved.

``` r
styler::style_text(
  "#' @export
  
  # bla bla bla
  foo <- function() NULL"
)
#> #' @export
#> 
#> # bla bla bla
#> foo <- function() NULL
```

Curious to hear what you think.

**Comment by MichaelChirico** [comment]
I see this a lot too and regular favor styler removing it by default . I think of it like trailing_blank_lines_linter for roxygen blocks

**Comment by lorenzwalthert** [comment]
I agree we could remove blank lines after `#'`. This could be a new transformer in `R/rules-line-break.R`, and it could be off for `strict = FALSE`. @IndrajeetPatil interested in a PR? I can provide guidance. You can see how rules were added previously, e.g. https://github.com/r-lib/styler/pull/569/files.

**Comment by IndrajeetPatil** [comment]
> @IndrajeetPatil interested in a PR? I can provide guidance.

Let me give this a try! And, yes, I will definitely need your guidance 🙃

> You can see how rules were added previously, e.g. https://github.com/r-lib/styler/pull/569/files

That's a good starting point. Thanks.
